third_party/ibis/ibis_teradata/__init__.py

```diff
@@ -141,7 +141,7 @@ def get_schema(self, table_name: str, database: str = None) -> sch.Schema:
         return sch.Schema(schema)
 
     def _get_schema_using_query(self, query):
-        cur = self.raw_sql(query)
+        cur = self.raw_sql(f"SELECT TOP 1 * FROM ({query}) AS t0")
```

Comment:
We might also have this problem for other engines, e.g: Hive:
```
def _get_schema_using_query(self, query):
    # Removing LIMIT 0 around query since it returns no results in Hive
    cur = self.raw_sql(query)
    cur.fetchall()
    ibis_fields = self._adapt_types(cur.description)
    cur.release()

    return sch.Schema(ibis_fields)
```

We should be able to use `fetchone` in place of `fetchall` for this. Or embed in a subquery.  I didn't check all engines so Hive may not be the only one.

Happy for this to be spun into another issue if you need to move quickly on this one.

Reply:
Good catch! I'll update this for Hive and add a `LIMIT 1` similar to the Ibis impala connection. Spanner only materializes one row, and all other DBs are handled by Ibis.